lang/csharp/src/apache/test/Schema/SchemaTests.cs

```diff
@@ -439,5 +439,57 @@ public void TestUnionSchemaWithoutTypeProperty(string schemaJson, string expecte
             var schema = Schema.Parse(schemaJson);
             Assert.AreEqual(schema.ToString(), expectedSchemaJson);
         }
+
+        [TestFixture]
+        public class SchemaTypeExtensionsTests
+        {
+            [TestCase("null", Schema.Type.Null)]
+            [TestCase("boolean", Schema.Type.Boolean)]
+            [TestCase("int", Schema.Type.Int)]
+            [TestCase("long", Schema.Type.Long)]
+            [TestCase("float", Schema.Type.Float)]
+            [TestCase("double", Schema.Type.Double)]
+            [TestCase("bytes", Schema.Type.Bytes)]
+            [TestCase("string", Schema.Type.String)]
+            [TestCase("record", Schema.Type.Record)]
+            [TestCase("enumeration", Schema.Type.Enumeration)]
+            [TestCase("array", Schema.Type.Array)]
+            [TestCase("map", Schema.Type.Map)]
+            [TestCase("union", Schema.Type.Union)]
+            [TestCase("fixed", Schema.Type.Fixed)]
+            [TestCase("error", Schema.Type.Error)]
+            [TestCase("logical", Schema.Type.Logical)]
+            [TestCase("Logical", null)]
+            [TestCase("InvalidValue", null)]
+            [TestCase("\"null\"", null)]
+            [TestCase("", null)]
+            [TestCase(null, null)]
+            public void ParseTypeTest(string value, object expectedResult)
+            {
+                if (expectedResult is Schema.Type expectedType)
+                {
+                    Assert.AreEqual(Schema.ParseType(value), expectedType);
+                }
+                else
+                {
+                    Assert.AreEqual(Schema.ParseType(value), expectedResult);
+                }
+            }
+
+            [TestCase("\"null\"", Schema.Type.Null)]
+            [TestCase("\"nu\"ll\"", null)]
+            [TestCase("\"\"", null)]
+            public void ParseTypeRemoveQuotesTest(string value, object expectedResult)
```

Comment:
Using the existing pattern.  If we want to change the pattern I would recommend a separate story with subtasks to update all the existing test files.

Reply:
In this case keeping previous patterns dont really matter. The test code iotself seems to me over complicated. I really do prefer to use good patterns where it can live side by side with existing code patterns

Reply:
I am not sure why, but my github showed this resolved and it should not. I might have clicked the wrong button.

Reply:
Again, If we are changing the testing pattern of how the cases are set up, I would rather it be a separate PR.

Reply:
This test is too compliacted as it is since it ahoild be a one liner. Please make the changes. If hou see where unit tests can be clarified more and there is a advantage to rework existing tests, that is absolutely fine. However new code should be optimal IMO.

Reply:
Now I see what you were saying.  It's been a while since I could trust a test frameworks Equals method.